### ISSUE TITLE
Code return warning on rm|ls|cp|mv to prevent "Argument list too long"

### ISSUE THREAD

**Issue by Jambon1510** [koalaman/shellcheck#2802]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
mv "$myfolder"\.* .

```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

shellcheck should warn about that in case of a potential "Argument list too long", a code check should be done or xargs should be used (https://unix.stackexchange.com/questions/128559/solving-mv-argument-list-too-long) for related commands (rm|ls|cp|mv)

**Comment by aeiplatform** [comment]
ShellCheck isn't Shell interpreter so how could it know?
Use find if you want to prevent error.

**Comment by e-kwsm** [comment]
This is not specific to rm, ls, cp, and mv, and this looks overzealous.
